### ISSUE TITLE
fix(router): handle navigating in the middle of another navigation

### ISSUE THREAD

**Issue by btford** [angular/angular#2776]
Behavior of rerouting during a navigation:
- [ ] before any components are committed, cancel and start over (recognition, canActivate, or loadResources/resolve) phases
- [ ] if any components are committed (any phases after load/resolve), then finish current nav before beginning the next

**Comment by naomiblack** [comment]
this is blocked by #2640

**Comment by naomiblack** [comment]
@btford can you check whether effort I assigned seems reasonable? (If not, please update)

**Comment by btford** [comment]
Seems reasonable. I'm moving this to the backlog since there are so many other things on my plate.

**Comment by btford** [comment]
As part of my work on #2640, I've changed the behavior for now to always queue up the next navigation after the current one finishes.

**Comment by mhevery** [comment]
This is obsolete, and no longer relevant / actionable. To keep our issues clean, we are aggressively closing them.

Covered by the new router.
